src/app/record/object-record/object-record.component.ts

```diff
@@ -0,0 +1,65 @@
+import { Component, OnInit, Input } from '@angular/core';
+import { JsonPatch } from '../../shared/interfaces';
+
+@Component({
+  selector: 'object-record',
+  templateUrl: './object-record.component.html',
+  styleUrls: ['./object-record.component.scss']
+})
+export class ObjectRecordComponent implements OnInit {
+  @Input() record: any;
+  @Input() diff: JsonPatch[];
+  @Input() depth: number;
+  @Input() expanded: boolean;
+  keys: Array<string> = [];
+  diffKeys = [];
+  newElement: any;
+  diffType: string;
+  private isArray = false;
+  value: any;
+  expandedArray: boolean[] = [];
+  constructor() { }
+
+  ngOnInit() {
+    this.keys = Object.keys(this.record);
+    if (this.diff) {
+      this.diff.forEach(item => {
+        if (item.path.length > this.depth) {
+          this.diffKeys.push(item.path[this.depth]);
+          this.expand(item.path[this.depth]);
+        } else if (item.path.length <= this.depth) {
+          this.diffKeys = this.keys;//expand the added object
```

Comment:
`this.keys` always empty?

Reply:
The keys are assigned here 'this.keys = Object.keys(this.record); '